src/server_manager/web_app/app.ts

```diff
@@ -652,24 +657,26 @@ export class App {
   private setServerManagementView(server: server.Server): void {
     // Show view and initialize fields from selectedServer.
     const view = this.appRoot.getServerView(server.getId());
+    const version = server.getVersion();
     view.selectedPage = 'managementView';
-    view.serverId = server.getMetricsId();
+    view.serverId = server.getId();
+    view.metricsId = server.getMetricsId();
     view.serverName = server.getName();
     view.serverHostname = server.getHostnameForAccessKeys();
     view.serverManagementApiUrl = server.getManagementApiUrl();
     view.serverPortForNewAccessKeys = server.getPortForNewAccessKeys();
     view.serverCreationDate = server.getCreatedDate();
-    view.serverVersion = server.getVersion();
-    view.accessKeyDataLimit = dataLimitToDisplayDataAmount(server.getAccessKeyDataLimit());
-    view.isAccessKeyDataLimitEnabled = !!view.accessKeyDataLimit;
+    view.serverVersion = version;
+    view.defaultDataLimitBytes = server.getDefaultDataLimit()?.bytes;
+    view.isDefaultDataLimitEnabled = view.defaultDataLimitBytes !== undefined;
     view.showFeatureMetricsDisclaimer = server.getMetricsEnabled() &&
-        !server.getAccessKeyDataLimit() && !hasSeenFeatureMetricsNotification();
+        !server.getDefaultDataLimit() && !hasSeenFeatureMetricsNotification();
 
-    const version = server.getVersion();
     if (version) {
       view.isAccessKeyPortEditable = semver.gte(version, CHANGE_KEYS_PORT_VERSION);
```

Comment:
Agreed, although this is all sort of glue code anyways.  If we're going to do that maybe we replace most of this method with a `.showServer(server)` method on the view.  That would break our "don't leak server details into the UI" rule though.  It's worth thinking about this.

Reply:
My point is to hide how we determine if a feature is available. The `Server` model would have methods like `isAccessKeyPortEditable()`
All those version checks should be hidden. The user of Server shouldn't have to be exposed to that, it's an implementation detail.